### ISSUE TITLE
service.running fails on CentOS 7

### ISSUE THREAD

**Issue by asemlef** [saltstack/salt#29064]
I just installed the development branch of Salt on a fresh [CentOS 7 instance in AWS](https://aws.amazon.com/marketplace/pp/B00O7WM7QW) and I get the following error when attempting to run a simple service.running state for mariadb:
```
[INFO    ] Running state [mariadb] at time 22:19:33.890532
[INFO    ] Executing state service.running for mariadb
[INFO    ] Executing command 'systemctl --full --no-legend --no-pager list-units mariadb.service' in directory '/root'
[ERROR   ] Command 'systemctl --full --no-legend --no-pager list-units mariadb.service' failed with return code: 1
[ERROR   ] stderr: Too many arguments.
[ERROR   ] retcode: 1
[INFO    ] Executing command 'systemctl --full --no-legend --no-pager list-unit-files mariadb.service' in directory '/root'
[ERROR   ] Command 'systemctl --full --no-legend --no-pager list-unit-files mariadb.service' failed with return code: 1
[ERROR   ] stderr: Invalid number of arguments.
[ERROR   ] retcode: 1
[ERROR   ] The named service mariadb is not available
[INFO    ] Completed state [mariadb] at time 22:19:33.901203 duration_in_ms=10.671
```

I've tested with other states and it appears to happen with all services, not just mariadb. From the looks of it the systemctl command it's using to check if the service exists is malformed and always erroring. Here is my salt version:
```
Salt Version:
           Salt: 2015.8.2-2992-g30d0e7a

Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
         Python: 2.7.5 (default, Jun 24 2015, 00:41:19)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed

System Versions:
           dist: centos 7.1.1503 Core
        machine: x86_64
        release: 3.10.0-229.14.1.el7.x86_64
         system: CentOS Linux 7.1.1503 Core
```

**Comment by terminalmage** [comment]
This was broken in 17b4538. I've pinged the author of that commit.

**Comment by aboe76** [comment]
systemd.py should be able to use the following commands without iterating at all:

for  get_enabled: 
systemctl is-enabled <name of service>

output will be 'enabled' and  exists with exitcode 0
output will be 'disabled' and exitcode 1 will be used

if no service file exists: exitcode 1 will be used with the following output:
Failed to get unit file state for syslog.service: No such file or directory

**Comment by julianbrost** [comment]
@aboe76 `systemctl is-enabled` doesn't work on Debian for services that don't have a systemd unit file, but are generated by `systemd-sysv-generator`. See for example:

    $ systemctl status autofs
    ● autofs.service - LSB: Automounts filesystems on demand
       Loaded: loaded (/etc/init.d/autofs)
       Active: active (running) since Tue 2015-10-13 23:19:35 CEST; 1 months 9 days ago
    [...]
    $ systemctl is-enabled autofs
    Failed to get unit file state for autofs.service: No such file or directory


Can `systemctl cat` be used for this? `systemctl cat autofs` will print the autogenerated unit file while `systemctl cat doesnotexist` will just output nothing. It looks like it also correctly handles template units, i.e. `systemctl cat openvpn@foobar` just works.

**Comment by aboe76** [comment]
@julianbrost that will only work to see if a service file exists, before checking the status.

I didn't think of the non existant service file as a init file, I tested it on a pure systemd operating system without sysvinit..

**Comment by terminalmage** [comment]
``systemctl is-enabled`` is not present in older systemd releases, so we cannot assume it will work.

**Comment by aboe76** [comment]
@terminalmage thanks for that info, didn't know that. so speeding up systemd module will be difficult.

**Comment by terminalmage** [comment]
@aboe76 If we can ascertain when it was added, we can conditionally use ``systemctl is-enabled``. I'll see what I can find.

**Comment by aboe76** [comment]
@terminalmage thanks for looking into this.

**Comment by terminalmage** [comment]
@aboe76 ``systemctl is-enabled`` seems to have been around since quite early, I found references to it for as early as systemd 24 (current release is ~227).

So, I take back what I said earlier, we should be safe in using it. I must have been confusing it with something else.

**Comment by k00mi** [comment]
@aboe76 @terminalmage You're welcome to develop your suggestions into a fix, I don't use CentOS and don't have time to come up with something that works on all systems. Just pretty please don't iterate over all unit files in every case.

**Comment by terminalmage** [comment]
This is fixed in #29830.
